ember_debug/object-inspector.js

```diff
@@ -22,6 +22,18 @@ try {
 
 const keys = Object.keys || Ember.keys;
 
+/**
+ * workaround to support detection of `[object AsyncFunction]` as a function
+ * @param value
+ * @returns {string}
+ */
+function typeOf(value) {
```

Comment:
@patricklx what did we need this change for?

Reply:
Ember.typeOf will not detect async function as function and will tell its an object instead.
I could also make a pr against emberjs for that...